delta/internal/pprof/stub.go

```diff
@@ -0,0 +1,9 @@
+package pprof
+
+import _ "unsafe"
+
+//go:linkname runtime_expandFinalInlineFrame runtime/pprof.runtime_expandFinalInlineFrame
+func runtime_expandFinalInlineFrame(stk []uintptr) []uintptr
+
+//go:linkname runtime_cyclesPerSecond runtime/pprof.runtime_cyclesPerSecond
+func runtime_cyclesPerSecond() int64
```

Comment:
It seems both functions are pretty "safe", meaning that even if the contract change, it should not harm the application that imports the client package. However, consider using build constraint on Go version

Reply:
I've added build tags, also added a unit test for signature comparision
https://github.com/pyroscope-io/godeltaprof/pull/1/files#diff-414cede1f0022eaa99267bf9e211d7fa3d84a74f101b4b9fc3d8030519876006

Reply:
I actually argue they are not that safe, imagine a signature changes like this:

```
func foo() (int, []uintptr) {
	return 0xcafebabe, []uintptr{0, 0, 0}
}

//go:linkname bar github.com/pyroscope-io/godeltaprof/internal/pprof.foo
func bar() []uintptr

```
```
[signal SIGSEGV: segmentation violation code=0x1 addr=0xcafebabe pc=0x46a109]
```

Also depending on the size and complexity of the target function, it may want to save arguments from registers to stack and it is callers responsibility to allocate the stack and if the signature changes we may overflow smth on the stack

Reply:
I won't argue about about safety – any link to runtime internals is unsafe. A change of the ABI (for any supported arch) or signature, renaming or removal of the `runtime_*` function, etc: everything breaks compatibility; therefore I suggested use of build constraint

Reply:
yep, totally agree, added buildtags, added a unit test to detect signature change